lib/SILOptimizer/IPO/GlobalOpt.cpp

```diff
@@ -770,6 +786,35 @@ void SILGlobalOpt::optimizeInitializer(SILFunction *AddrF,
   HasChanged = true;
 }
 
+/// Look through all the uses of the global variable. If the only use is a
+/// store, and the global is private, then we can remove the global, the store,
+/// and the alloc. Otherwise, return.
+void SILGlobalOpt::tryOptimizeUnusedGlobal(SILGlobalVariable *global,
+                                           StoreInst *store) {
+  if (global->getLinkage() != SILLinkage::Private)
+    return;
+  if (GlobalLoadMap.count(global) || GlobalAccessMap.count(global))
+    return;
+
+  auto globalAddr = cast<GlobalAddrInst>(store->getDest());
+  bool canRemove = true;
+  for (auto *use : globalAddr->getUses()) {
```

Comment:
This loop only looks at uses within this function, so if you have a global map of all non-store uses, it shouldn't be necessary at all.

Reply:
I removed this while addressing your below comment.